app/controllers/api/alert_definitions_controller.rb

```diff
@@ -1,16 +1,13 @@
 module Api
   class AlertDefinitionsController < BaseController
-    REQUIRED_FIELDS = %w(description db expression options).freeze
-
     before_action :set_additional_attributes
 
     def create_resource(type, id, data = {})
       assert_id_not_specified(data, type)
-      assert_all_required_fields_exists(data, type, REQUIRED_FIELDS)
       begin
-        data["expression"] = MiqExpression.new(data["expression"])
-        data["enabled"] = true if data["enabled"].nil?
-        super(type, id, data).serializable_hash.merge("expression" => data["expression"])
+        update_miq_expression(data) if data["expression"]
+        alert = super(type, id, data.deep_symbolize_keys).serializable_hash
```

Comment:
I'm concerned that this `deep_symbolize_keys` will corrupt an incoming `MiqExpression`:

```ruby
{"=" => {"field" => "foo", "value" => "bar"}}.deep_symbolize_keys # => {:"=" => {:field => "foo", :value => "bar"}}
```